Add a constructor to SqsMessageHandler which accepts a DynamicQueueUr…

sethkelly opened the pull request:
…lDestinationResolver.

Specifically, I would like the ability to set the DynamicQueueUrlDestinationResolver.autoCreate flag so that the queue specified in my SqsMessageHandler can be auto-created.

pivotal-issuemaster commented:
@sethkelly Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-integration-aws&pullRequestId=96)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-integration-aws&pullRequestId=96) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

artembilan commented:
@sethkelly ,

Thank you for the report, but before we go for review you need to sign the CLA: https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-integration-aws&pullRequestId=96

pivotal-issuemaster commented:
@sethkelly Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-integration-aws&pullRequestId=96)!

sethkelly commented:
Sorry, I see the test failure my commit added, in an unexpected test class.  Looking into it now.

artembilan commented on src/main/java/org/springframework/integration/aws/outbound/SqsMessageHandler.java:
```diff
@@ -75,12 +75,16 @@
+		this(amazonSqs, new DynamicQueueUrlDestinationResolver(amazonSqs, resourceIdResolver));
+	}
+
+	public SqsMessageHandler(AmazonSQSAsync amazonSqs, DynamicQueueUrlDestinationResolver destinationResolver) {
```
Please, consider to extend it to be based on the `DestinationResolver<String>` interface instead.

Also there should be an `Assert.notNull(destinationResolver, "destinationResolver must not be null");` do not let to inject `null`.

sethkelly replied:
Seems I should also add a null check on amazonSqs?

artembilan replied:
Yes, please 😀

sethkelly commented:
I've added a test which validates my desired use case (auto-creation of queues).  I've also added input null checking on the constructor and marked myself as an author.  Looks like the kinesis test is indeed unrelated and an intermittent failure.  Let me know if you'd like any other changes.

artembilan commented:
@sethkelly ,

If you are interested I fixed that test like this: https://github.com/spring-projects/spring-integration-aws/commit/857d184ccac58629c9d365946c252696af87ac5d
